include/openenclave/edger8r/common.h

```diff
@@ -66,14 +66,24 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
     return result;
 }
 
-#define OE_ADD_SIZE(total, size)                \
-    do                                          \
-    {                                           \
-        if (oe_add_size(&total, size) != OE_OK) \
-        {                                       \
-            _result = OE_INTEGER_OVERFLOW;      \
-            goto done;                          \
-        }                                       \
+#define OE_ADD_SIZE(total, size)                                 \
+    do                                                           \
+    {                                                            \
+        if (sizeof(total) > sizeof(size_t) && total > SIZE_MAX)  \
+        {                                                        \
+            _result = OE_INVALID_PARAMETER;                      \
+            goto done;                                           \
+        }                                                        \
+        if (sizeof(size) > sizeof(size_t) && size > SIZE_MAX)    \
```

Comment:
It seems very weird to me to compare both "size" and "sizeof(size)" here, so I've been staring at this for like a good 20 minutes thinking about what this means (it's the holidays... my brain still isn't working quite right yet).

But I think I know what we're doing here. Basically something like: "If someone has tried to invoke this macro with parameters that are larger SIZE_MAX, then error out".

I started thinking like "is that second condition even possible? how would someone trigger that second comparison to execute, considering the comparison `size > SIZE_MAX` needs to be evaluated on numeric values, are there any numeric types for `size` that could lead to this?" and yes, there's the notorious `long double`, our best friend.

This change looks reasonable to me!

Reply:
If sizeof(size) is larger than size_t but its value is still less than `SIZE_MAX` then we can still allow the cast.

On 64-bit machines size_t will be 64 bits and I don't know of any numeric types which are larger than that. And since SGX is only available on 64-bit processors (as far as I know) the case will probably never be hit.

But if we somehow end up running on a 32-bit CPU then and 64 bit type like `long long` would cause the second condition to be evaluated.